### ISSUE TITLE
UuidValue, internal value can be uppercase.

### ISSUE THREAD

**Issue by giorgiofran** [daegalus/dart-uuid#95]
Hi, moving to 4.0.0 I found a problem that, if I create an UuidValue instance with a string with upper cases, it is not converted to lower cases.
From what I can see, the class assumes the internal value to be lowercase, also looking at the compare method.
For what I can see the problem is here:

 ` ` ` dart
const UuidValue(this.uuid);
` ` `

probably it should be something like
 ` ` ` dart
const UuidValue(String uuid) : uuid = uuid.toLowerCase();
` ` `

**Comment by daegalus** [comment]
Good catch. I will do some testing and make the necessary fixes.

**Comment by daegalus** [comment]
Ok, I did some testing. We can't enforce anything on the default const constructor. Since it is const, we cant have a body, and we can't do any manipulation.

So I have made the default const constructor private, and added a `UuidValue.fromString()` factory constructor. It just does a `toLowerCase()` before passing it to the UuidValue private const constructor. Same logic as `UuidValue.withValidation()` but without the validation step.

I will be releasing 4.1.0 shortly.
